llvm/lib/Target/WebAssembly/WebAssemblyLowerEmscriptenEHSjLj.cpp

```diff
@@ -198,9 +198,18 @@
 ///
 /// If there are calls to setjmp()
 ///
-/// 2) and 3): The same as 2) and 3) in Emscripten SjLj.
-/// (setjmpTable/setjmpTableSize initialization + setjmp callsite
-/// transformation)
+/// 2) In the function entry that calls setjmp, initialize
+///    functionInvocationId as follows:
+///
+///    functionInvocationId = alloca()
```

Comment:
Can you include the size that passed to alloca here?  

 (looks like its getInt32Ty() so maybe put 4 here?)

Reply:
Is the same functionInvocationId used even if there are several calls to setjmp inside the same function?

Reply:
Yes. The combination of function invocation ID + callsite determines the unique `setjmp` call, in my understanding.

Reply:
yes